packages/core/test/unit/query-generator/add-index-query.test.ts

```diff
@@ -0,0 +1,282 @@
+      default: 'CREATE INDEX [index_1] ON [table] ([column1], [column2])',
+      snowflake: notSupportedError,
+      'mariadb mysql': 'ALTER TABLE `table` ADD INDEX `index_1` (`column1`, `column2`)',
+    });
+  });
+
+  it('generates a query that adds a unique index with a name', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: ['column1', 'column2'], name: 'index_1', unique: true }), {
+      default: 'CREATE UNIQUE INDEX [index_1] ON [table] ([column1], [column2])',
+      ibmi: `BEGIN DECLARE CONTINUE HANDLER FOR SQLSTATE VALUE '42891' BEGIN END; ALTER TABLE "table" ADD CONSTRAINT index_1 UNIQUE ("column1", "column2"); END`,
+      snowflake: notSupportedError,
+      'mariadb mysql': 'ALTER TABLE `table` ADD UNIQUE INDEX `index_1` (`column1`, `column2`)',
+    });
+  });
+
+  it('generates a query that adds an index concurrently', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: ['column1', 'column2'], name: 'index_1', concurrently: true }), {
+      default: buildInvalidOptionReceivedError('addIndexQuery', dialect.name, ['concurrently']),
+      postgres: 'CREATE INDEX CONCURRENTLY "index_1" ON "table" ("column1", "column2")',
+      snowflake: notSupportedError,
+    });
+  });
+
+  it('generates a query that adds an index if it doesnt already exist', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: ['column1', 'column2'], name: 'index_1', ifNotExists: true }), {
+      default: buildInvalidOptionReceivedError('addIndexQuery', dialect.name, ['ifNotExists']),
+      mariadb: 'ALTER TABLE `table` ADD INDEX IF NOT EXISTS `index_1` (`column1`, `column2`)',
+      snowflake: notSupportedError,
+      'postgres sqlite': 'CREATE INDEX IF NOT EXISTS [index_1] ON [table] ([column1], [column2])',
+    });
+  });
+
+  it('generates a query that adds an index with included columns', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: ['column1', 'column2'], name: 'index_1', include: ['column3'] }), {
+      default: buildInvalidOptionReceivedError('addIndexQuery', dialect.name, ['include']),
+      db2: new Error('DB2 does not support non-unique indexes with INCLUDE syntax.'),
+      snowflake: notSupportedError,
+      'mssql postgres': 'CREATE INDEX [index_1] ON [table] ([column1], [column2]) INCLUDE ([column3])',
+    });
+  });
+
+  it('generates a query that adds a unique index with included columns', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: ['column1', 'column2'], name: 'index_1', include: ['column3'], unique: true }), {
+      default: buildInvalidOptionReceivedError('addIndexQuery', dialect.name, ['include']),
+      snowflake: notSupportedError,
+      'db2 mssql postgres': 'CREATE UNIQUE INDEX [index_1] ON [table] ([column1], [column2]) INCLUDE ([column3])',
+    });
+  });
+
+  it('generates a query that adds a unique index with included columns using an array of liternals', () => {
```

Comment:
should test against multiple type of expressions (`sql.x`) as there was a mistake in the code that limited it to literals

Reply:
Updated test now